### PR TITLE
Improve dragging behaviour when a scale transform is applied.

### PR DESCRIPTION
**Pull request by matrushka**
Has a lot of changes due to code formatting but main changes are in this commit: https://github.com/PuruVJ/svelte-drag/commit/242b30b824a5851947d02e165b015c7e706da132

I also removed the `previousX` and `previousY` variables as they were assigned to 0 and not reassigned at all. Using `initialX` and `initialY` instead of them in the snapping logic worked well.

### CODE REVIEW

**Comment by PuruVJ** [comment]
Help me understand this: Why scale is needed in the first place? Is there some unexpected behavior that happens without it?

File: src/index.ts [hunk 1]
```diff
@@ -241,356 +241,363 @@ export type DragOptions = {
...
+		const { clientX, clientY } = isTouchEvent(e) ? e.touches[0] : e;
 
-    const { clientX, clientY } = isTouchEvent(e) ? e.touches[0] : e;
+		// Get final values for clamping
+		let [finalX, finalY] = [clientX, clientY];
 
-    // Get final values for clamping
-    let [finalX, finalY] = [clientX, clientY];
+		// Calculate the current scale of the node
+		let scale = node.offsetWidth / nodeRect.width;
```
Comment (PuruVJ) [inline src/index.ts:439]: Can you explain what this line does? What's the difference between offsetWidth and nodeRect.width?

Comment (matrushka) [reply src/index.ts:439]: > Help me understand this: Why scale is needed in the first place? Is there some unexpected behavior that happens without it?

Imagine that I have a canvas where I have draggable elements on top of it (let's say we have element A 100px by 100px), and I want to implement a zoom in and out function. I can add `transform: scale(2);` to zoom in 2x. But that breaks the calculation as it'll still change the translation by 100px but it actually needs to change it by 50px only as the scale is 2x. 

In the current behaviour the draggable element will either move too fast (if zoomed in) or move too slow (if zoomed out) compared to the cursor. This breaks the mouse events logic (mouseup happens outside the element) and also looks weird for the user.

`node.offsetWidth` gives you the unscaled width and the `nodeRect` gives you the transformed width. By using those I can calculate the scale to be used in calculation of translate values. When scale is added draggable element moves at the speed of the cursor when scaled, fixing the problems.

Comment (PuruVJ) [reply src/index.ts:439]: This is perfect! Though I'll be renaming scale to `inverseScale`, as scale to me feels like "the size compared to the original size", so just this!

Thanks for the PR

**Pull request merged**